src/Control/FunFlow/Steps.hs

```diff
@@ -64,3 +69,33 @@ retry n secs f = catch f $ proc (x, (_ :: ex)) -> do
   x1 <- pauseWith -< (secs,x)
   x2 <- retry (n-1) secs f -< x1
   returnA -< x2
+
+-- | @putFileInStore -< (inFile, outFile)@ copies the contents of @inFile@
+-- into the content store under the file name @outFile@.
+putFileInStore :: Flow eff ex (FileContent, FilePath) CS.Item
+putFileInStore = putInStore $ \dir (FileContent inPath, outFile) -> do
+  let outPath = dir </> outFile
+  copyFile inPath outPath
+
+-- | @putDirInStore -< (inDir, Just outDir)@ copies the contents of @inDir@
+-- into the content store under @outDir@.
+-- @putDirInStore -< (inDir, Nothing)@ copies the contents of @inDir@
+-- into the content store directly under the subtree.
+putDirInStore :: Flow eff ex (DirectoryContent, Maybe FilePath) CS.Item
+putDirInStore = putInStore $ \dir (DirectoryContent inPath, mbOutPath) -> do
+  let outPath = dir </> fromMaybe "" mbOutPath
```

Comment:
This is potentially a big source for bugs (see above comment about using `path`). `</>` has some possibly unexpected behaviour when its second argument is an absolute path:

```
λ> "/foo" </> "/bar"
"/bar"
```

While we're not hardening against deliberate attacks here, somebody could easily end up writing to a root dir accidentally just by adding an extra `/`

Reply:
I agree, that's too dangerous. `path` looks like it could be a good solution.